projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1615,6 +1673,18 @@ public function add_jetpack_social_og_images( $tags ) {
 		$opengraph_image = $this->get_social_opengraph_image( get_the_ID() );
 
 		if ( empty( $opengraph_image ) ) {
+			// If we do not have a SIG or attached image, but we have an image in post body
+			// we need to check that the image is not too large for the social sites.
+			if ( ! empty( $tags['og:image'] ) ) {
+				$reduced_image = $this->reduce_file_size( $tags['og:image'], $tags['og:image:width'], $tags['og:image:height'] );
+
+				if ( ! empty( $reduced_image ) ) {
+					$tags['og:image']        = $reduced_image['url'];
```

Comment:
I ran this branch on my test site and created this post with the same image that Matt used. https://sidpressable.mystagingwebsite.com/2024/01/22/ajn-22/

 The image does not load through in the link previews on Facebook/Mastodon. Here are the OG tags. 

<img width="1632" alt="Screenshot 2024-01-22 at 6 05 25 PM" src="https://github.com/Automattic/jetpack/assets/6594561/5f53103a-9f1f-4327-9047-e92be4eddd0e">




The og: image shared is this one:  `https://i0.wp.com/sidpressable.mystagingwebsite.com/wp-content/uploads/2024/01/era-300-hero-horizontal-white-3.jpg?w=8759.25&#038;h=4926.75&#038;ssl=1`, perhaps it should be formatted differently without the SSL bits in the end, possibly?

I have a feeling you're using the scaled down version of the original image on the local. Can you try uploading the [original image](https://assets.sonos.com/media-kits/hardware/era-300/era-300-hero-horizontal-white.jpg) to your local site and try again?

Reply:
I did use the original image; I double-checked.
Are you sure that the link preview is not working @spsiddarthan? I shared your post URL on Facebook, and it works fine for me like that.

![CleanShot 2024-01-22 at 13 47 06 png](https://github.com/Automattic/jetpack/assets/36671565/bd209088-bcea-42e5-bc8d-02c8b4a6ac0c)